crates/solver/src/settlement_submission/submitter/common.rs

```diff
@@ -26,7 +23,7 @@ impl PrivateNetwork {
     pub async fn submit_raw_transaction(
         &self,
         tx: TransactionBuilder<Web3Transport>,
-    ) -> Result<TransactionHandle, SubmitApiError> {
+    ) -> anyhow::Result<TransactionHandle> {
```

Comment:
ubernit: This file already `use anyhow::Result` so you should be able to:
```suggestion
    ) -> Result<TransactionHandle> {
```

Reply:
Tried it; didn't work.
I changed the API of the  trait from `-> Result` to `-> anyhow::Result` so at no place implementing the trait should the existing code be okay since the Results are different types.
Or am I missing something?

Reply:
My point is that `Result` should already have been `anyhow::Result` because we `use anyhow::Result` in the file (or at least I thought we did - but maybe I saw wrong).

Anyway, just an ubernit - so not blocking merge from my perspective.

Reply:
But if `Result` was already `anyhow::Result` the trait wouldn't have been correctly implemented before because the old trait expected `std::result::Result` instead of `anyhow::Result`. So either the signature was wrong before it's wrong now, no?

Reply:
This is confusing because anyhow::Result is defined like `type anyhow::Result<T> = std::result::Result<T, anyhow::Error>` so it is not a normal type. Writing `anyhow::Result<TransactionHandle>` and `std::result::Result<TransactionHandle, anyhow::Error>`(or leaving out the std::result part) is equivalent.

Reply:
Ahh, I see. Thanks for clearing that up. 👍

Reply:
There are also a few more places to replace. Personally, I don't favor any of the versions `anyhow::Result<TransactionHandle>` vs `Result<TransactionHandle>`, but I do favor using only one of them through out the code for consistency.